src/Data/List/Show/Core.agda

```diff
@@ -0,0 +1,279 @@
+  .sep   → "│"
+  .right → just "│"
+unicode .bot = just λ where
+  .left  → just "└"
+  .cont  → just '─'
+  .sep   → "┴"
+  .right → just "┘"
+
+ascii : Config
+ascii .top = just λ where
+  .left  → just "+"
+  .cont  → just '-'
+  .sep   → "-"
+  .right → just "+"
+ascii .sep = just λ where
+  .left  → just "|"
+  .cont  → just '-'
+  .sep   → "+"
+  .right → just "|"
+ascii .row = λ where
+  .left  → just "|"
+  .cont  → nothing
+  .sep   → "|"
+  .right → just "|"
+ascii .bot = just λ where
+  .left  → just "+"
+  .cont  → just '-'
+  .sep   → "-"
+  .right → just "+"
+
+compact : Config → Config
+compact c = record c { sep = nothing }
+
+module _ where
+
+  private
+    dropBorder : Line → Line
+    dropBorder l = record l { left = nothing; right = nothing }
+
+  noborder : Config → Config
+  noborder c .top = nothing
+  noborder c .sep = Maybe.map dropBorder (c .sep)
+  noborder c .row = dropBorder (c .row)
+  noborder c .bot = nothing
+
+module _ where
+
+  private
+    space : Line → Line
+    space l = let pad = maybe fromChar " " (l .cont) in λ where
+      .left  → Maybe.map (String._++ pad) (l .left)
+      .cont  → l .cont
+      .sep   → pad String.++ l .sep String.++ pad
+      .right → Maybe.map (pad String.++_) (l .right)
+
+  addSpace : Config → Config
+  addSpace c .top = Maybe.map space (c .top)
+  addSpace c .sep = Maybe.map space (c .sep)
+  addSpace c .row = space (c .row)
+  addSpace c .bot = Maybe.map space (c .bot)
+
+whitespace : Config
+whitespace .top = nothing
+whitespace .sep = nothing
+whitespace .row = λ where
+  .left  → nothing
+  .cont  → nothing
+  .sep   → " "
+  .right → nothing
+whitespace .bot = nothing
+
+-- /!\ Invariants:
+-- * the table is presented as a list of rows
+-- * header has the same length as each one of the rows
+--   i.e. we have a rectangular table
+-- * all of the strings in a given column have the same length
+
+table : Config → List (List String) → String
+table _ []              = ""
+table c (header ∷ rows) =
+  unlines $ map String.concat
+          $ th ++ (trs ∷ʳ? lbot) where
+
+  cellsOf : Maybe Char → List String → List String
+  cellsOf nothing  = id
+  cellsOf (just c) = map (λ cell → replicate (length cell) c)
+
+  lineOf : Line → List String → List String
+  lineOf l xs = l .left
+            ?∷  intersperse (l .sep) (cellsOf (l .cont) xs)
+            ∷ʳ? l .right
+
+  mlineOf : Maybe Line → List String → Maybe (List String)
+  mlineOf l xs = Maybe.map (λ l → lineOf l xs) l
+
+  ltop : Maybe (List String)
+  lsep : Maybe (List String)
+  tr   : List String → List String
+  lbot : Maybe (List String)
+
+  ltop = mlineOf (c. top) header
+  lsep = mlineOf (c. sep) header
+  tr   = lineOf (c. row)
+  lbot = mlineOf (c. bot) header
+
+  th  = ltop ?∷ tr header ∷ []
+  trs = if null rows then id else (maybe _∷_ id lsep)
+      $ maybe intersperse id lsep
+      $ map tr rows
+
+------------------------------------------------------------------------
+-- examples
```

Comment:
It might be better to lift these examples to `Data.List.Show` where more people are likely to find them?

Reply:
I think I will put them all in a README.